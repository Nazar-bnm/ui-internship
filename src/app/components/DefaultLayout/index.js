import React from 'react';
import { Route } from 'react-router-dom';

import Header from '../Header';
import Footer from '../Footer'

const DefaultLayout = ({component: Component, hideFooter, hideHeader, ...rest}) => {
  return (
    <Route {...rest} render={matchProps => (
      <div>
          {!hideHeader && <Header />}
          <Component {...matchProps} />
          {!hideFooter && <Footer />}
      </div>
    )} />
  )
};

export default DefaultLayout;
