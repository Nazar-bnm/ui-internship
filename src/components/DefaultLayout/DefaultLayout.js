import React from 'react';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';

import Header from '../Header';
import Footer from '../Footer';
import ProductImage from '../ProductImage';
import { images } from '../../config/productImageMockups';

const DefaultLayout = ({ component: Component, hideFooter, hideHeader, ...rest }) => {
  return (
    <Route {...rest} render={(matchProps) => (
      <div>
        {!hideHeader && <Header />}
        <Component {...matchProps} />
        <ProductImage imagesArr={ images }/>
        {!hideFooter && <Footer />}
      </div>
    )} />
  );
};

DefaultLayout.propTypes = {
  // Component which displayed as main content
  component: PropTypes.any,
  hideFooter: PropTypes.bool,
  hideHeader: PropTypes.bool,
};

export default DefaultLayout;
