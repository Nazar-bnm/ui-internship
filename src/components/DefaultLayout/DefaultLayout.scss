@import '../../scss/global_styles';