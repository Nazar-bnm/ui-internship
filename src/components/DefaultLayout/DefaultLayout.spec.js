import React from 'react';
import { shallow } from 'enzyme';
import DefaultLayout from './DefaultLayout';

describe('<DefaultLayout />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <DefaultLayout />
    );
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
