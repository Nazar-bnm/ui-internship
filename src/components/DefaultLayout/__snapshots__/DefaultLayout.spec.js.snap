// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultLayout /> should match snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
